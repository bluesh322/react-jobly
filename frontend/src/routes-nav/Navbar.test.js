import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import NavBar from "./Navbar";
import TestUserProvider from "../auth/testUser";

test("renders without crashing", () => {
  render(
    <MemoryRouter>
      <TestUserProvider
      >
        <NavBar></NavBar>
      </TestUserProvider>
    </MemoryRouter>
  );
});

test("matches snapshot logged in", async () => {
  const { asFragment } = render(
    <MemoryRouter>
      <TestUserProvider>
        <NavBar></NavBar>
      </TestUserProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot logged out", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <TestUserProvider user={null}>
        <NavBar></NavBar>
      </TestUserProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
