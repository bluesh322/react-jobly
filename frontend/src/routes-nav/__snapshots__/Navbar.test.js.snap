// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot logged in 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <div
        class="MuiContainer-root makeStyles-navDisplayFlex-3 MuiContainer-maxWidthMd"
      >
        <button
          aria-label="home"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <a
              href="/"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
                />
              </svg>
            </a>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <nav
          aria-labelledby="main navigation"
          class="MuiList-root makeStyles-navDisplayFlex-3 MuiList-padding"
        >
          <a
            class="makeStyles-linkText-4"
            href="/companies"
          >
            <div
              aria-disabled="false"
              class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                >
                  Companies
                </span>
              </div>
              <span
                class="MuiTouchRipple-root"
              />
            </div>
          </a>
          <a
            class="makeStyles-linkText-4"
            href="/jobs"
          >
            <div
              aria-disabled="false"
              class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                >
                  Jobs
                </span>
              </div>
              <span
                class="MuiTouchRipple-root"
              />
            </div>
          </a>
          <a
            class="makeStyles-linkText-4"
            href="/profile"
          >
            <div
              aria-disabled="false"
              class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                >
                  Profile
                </span>
              </div>
              <span
                class="MuiTouchRipple-root"
              />
            </div>
          </a>
          <a
            class="makeStyles-linkText-4"
            href="/logout"
          >
            <div
              aria-disabled="false"
              class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                >
                  Log Out
                </span>
              </div>
              <span
                class="MuiTouchRipple-root"
              />
            </div>
          </a>
        </nav>
      </div>
    </div>
  </header>
</DocumentFragment>
`;

exports[`matches snapshot logged out 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <div
        class="MuiContainer-root makeStyles-navDisplayFlex-5 MuiContainer-maxWidthMd"
      >
        <button
          aria-label="home"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <a
              href="/"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
                />
              </svg>
            </a>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <nav
          aria-labelledby="main navigation"
          class="MuiList-root makeStyles-navDisplayFlex-5 MuiList-padding"
        >
          <a
            class="makeStyles-linkText-6"
            href="/companies"
          >
            <div
              aria-disabled="false"
              class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                >
                  Companies
                </span>
              </div>
              <span
                class="MuiTouchRipple-root"
              />
            </div>
          </a>
          <a
            class="makeStyles-linkText-6"
            href="/jobs"
          >
            <div
              aria-disabled="false"
              class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                >
                  Jobs
                </span>
              </div>
              <span
                class="MuiTouchRipple-root"
              />
            </div>
          </a>
          <a
            class="makeStyles-linkText-6"
            href="/profile"
          >
            <div
              aria-disabled="false"
              class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                >
                  Profile
                </span>
              </div>
              <span
                class="MuiTouchRipple-root"
              />
            </div>
          </a>
          <a
            class="makeStyles-linkText-6"
            href="/logout"
          >
            <div
              aria-disabled="false"
              class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                >
                  Log Out
                </span>
              </div>
              <span
                class="MuiTouchRipple-root"
              />
            </div>
          </a>
        </nav>
      </div>
    </div>
  </header>
</DocumentFragment>
`;
